transcom/mymove
# B-17754 Added re-rendering action, and the count flag behavior.

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/TeamRoom.mvc/Show/848240)

## Summary

Is there anything you would like reviewers to give additional scrutiny?

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

Run the command
$: make db_dev_e2e_populate 
Go to office local:3000

Log into office local as a TOO
Look at the shipments
Change user and select 'Select Prime Simulator'
Look at the shipments
Locate an MTO shipment that is also listed under the prime simulator
Navigate to the MTO shipment
Click 'Add Service Item'
Set the service item type to origin sit, or destination sit.
Enter the details and click 'Create Service Item'
Find the shipment ID
Create a SIT extension on the swagger page or through postman using:
http://officelocal:3000/swagger-ui/prime.html#/mtoShipment/createSITExtension
With the shipment ID to reference for the endpoint.

Navigate to the MTO shipment as a TOO

<img width="1375" alt="Pasted Graphic 1" src="https://github.com/transcom/mymove/assets/148263228/bff99678-62a0-471e-aec0-fb37ef021c9e">

Navigate to the Move task order tab
<img width="1375" alt="Pasted Graphic 1" src="https://github.com/transcom/mymove/assets/148263228/5e258c57-b79a-4a0f-a6d8-dee8df5dd157">

Observe
- The HHG Shipment entry on the left side bar
- The Move task order entry on the top bar
<img width="424" alt="Pasted Graphic 2" src="https://github.com/transcom/mymove/assets/148263228/26b71beb-4b8b-4cb4-8f30-79a112c60d8f">

Navigate (either by clicking HHG shipment, or scroll down) to the Household goods section.

Find and click the Review request element
<img width="1253" alt="Pasted Graphic 3" src="https://github.com/transcom/mymove/assets/148263228/1b2843ad-8168-46bd-acd4-4f94cc1b4859">

Type in a number in the Total days of SIT approved text box, but the value must be greater than the set value. (IE: 91 instead of 90)

Check Yes in the radio elements for Accept request for extension.

Submit the form (Either by pressing enter, or clicking  Save)
<img width="705" alt="Pasted Graphic 4" src="https://github.com/transcom/mymove/assets/148263228/b147363f-8e31-4dea-999f-6e0320e6c334">

You should see the yellow warning flags decrement by 1.

To start with… This scenario should display:
HHG Shipment flag with 4 warnings and the Move task order flag with 5 warnings.

The warning for excess weight is tracked within the Move task order flags.
As the warnings are cleared, you should observe the flags reduce warning flags by that much.

The page should not need to be refreshed to observe these changes
 

### Frontend


## Screenshots
